src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -25,24 +25,35 @@
 package org.spongepowered.api;
 
 /**
- * Represents a type of a catalog that can be used to identify types
- * without using an {@link Enum}.
+ * Represents a type of a catalog that can be used to identify types without
+ * using an {@link Enum}.
+ *
+ * <p>Implementing classes must meet the following requirements for all its
+ * members including those which are not listed in the classes listed in the
+ * {@link org.spongepowered.api.util.annotation.CatalogedBy CatalogedBy}
+ * annotation: <ul> <li>Every pair (a, b) that matches ... <ul>
+ * <li>a.equals(b) must also match a == b</li>
+ * <li>a.getId().equalsIgnoreCase(b.getId()) must also match a == b</li> </ul>
+ * </li> </ul> </p>
  */
 public interface CatalogType {
 
     /**
-     * Gets the unique identifier of this {@link CatalogType}. The identifier
-     * can be formatted however needed.
+     * Gets the unique identifier of this {@link CatalogType}. Its uniqueness is
+     * guaranteed to the include every character case. The id of this instance
```

Comment:
Perhaps something like 'Its uniqueness is guaranteed regardless of character case'